net/datacenter-node-install/scripts/set-hostname.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+if [[ "$EUID" -ne 0 ]]
+  then echo "Please run as root"
+  exit
+fi
+
+set -xe
+
+sed -i 's/preserve_hostname: false/preserve_hostname: true/' /etc/cloud/cloud.cfg
```

Comment:
This should be done via a drop-in to avoid the inevitable merge conflict during `apt upgrade` when vendor defaults change.  Something like
```
echo "preserve_hostname: false" > /etc/cloud.cfg.d/99-disable-preserve-hostname.cfg
```